src/Compilers/CSharp/Portable/CommandLine/CSharpCommandLineParser.cs

```diff
@@ -1538,16 +1578,16 @@ public new CSharpCommandLineArguments Parse(IEnumerable<string> args, string? ba
             };
         }
 
-        private static void ParseAndResolveReferencePaths(string? switchName, string? switchValue, string? baseDirectory, List<string> builder, MessageID origin, List<Diagnostic> diagnostics)
+        private static void ParseAndResolveReferencePaths(string? switchName, ReadOnlyMemory<char>? switchValue, string? baseDirectory, List<string> builder, MessageID origin, List<Diagnostic> diagnostics)
         {
-            if (string.IsNullOrEmpty(switchValue))
+            if (switchValue is not { Length: > 0 })
             {
                 RoslynDebug.Assert(!RoslynString.IsNullOrEmpty(switchName));
                 AddDiagnostic(diagnostics, ErrorCode.ERR_SwitchNeedsString, MessageID.IDS_PathList.Localize(), switchName);
                 return;
             }
 
-            foreach (string path in ParseSeparatedPaths(switchValue))
+            foreach (string path in ParseSeparatedPaths(switchValue.Value.ToString()))
```

Comment:
Do we want to get rid of the iterator allocation here also?

Reply:
We certainly could but this is not really a hot path, or even a common path, for C# compilation. To keep the churn down I'd been focusing on areas of the argument parsing that hit hot paths. This particular argument doesn't get hit at all in modern MSBuild driven scenarios that I'm aware of.

I'm certainly happy to push further here if you think it's worth the code churn.